types/time.go

```diff
@@ -1712,6 +1712,17 @@ func ParseDate(sc *stmtctx.StatementContext, str string) (Time, error) {
 	return ParseTime(sc, str, mysql.TypeDate, MinFsp)
 }
 
+// ParseTimeFromYear parse a `YYYY` formed year to corresponded Datetime type.
+// year must be in the range [MinYear, MaxYear].
```

Comment:
change the comment to:
Note: the invoker must promise the \`year\` is in the range [MinYear, MaxYear]